stock_picking_type_bypass_reservation/models/stock_move.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2025 Moduon Team S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0)
+
+from odoo import fields, models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def _should_bypass_reservation(self, forced_location=False):
+        res = super()._should_bypass_reservation(forced_location=forced_location)
+        return res or self.env.context.get("force_bypass_reservation")
+
+    def _action_assign(self, force_qty=False):
+        # Do a first run to gather items in stock and then bypass the reservation for
+        # the rest. FIXME: This won't work for items without tracking as the first
+        # reserved line will be set to update with the whole move demand.
+        res = super()._action_assign(force_qty=force_qty)
+        moves_to_test_bypass = self.filtered(
+            lambda x: x.picking_type_id
+            and x.picking_type_id.should_bypass_reservation()
+        )
+        if moves_to_test_bypass:
+            moves_to_test_bypass.with_context(force_bypass_reservation=True)
```

Comment:
Unnecessary

```suggestion
```

I prefer the old way without the method call `should_bypass_reservation`. Maybe we can delete this function (less code and maybe we don't override never. And if we do it, an inheritance must be done... and on master branch this function is not present)